gcloud-java-compute/src/main/java/com/google/gcloud/compute/InstanceInfo.java

```diff
@@ -0,0 +1,1282 @@
+  }
+
+  /**
+   * Returns a list of service accounts, with their specified scopes, authorized for this instance.
+   * Service accounts generate access tokens that can be accessed through the metadata server and
+   * used to authenticate applications on the instance.
+   *
+   * @see <a href="https://cloud.google.com/compute/docs/authentication">Authenticating from
+   *     Google Compute Engine</a>
+   */
+  public List<ServiceAccount> serviceAccounts() {
+    return serviceAccounts;
+  }
+
+  /**
+   * Returns the scheduling options for the instance.
+   */
+  public SchedulingOptions schedulingOptions() {
+    return schedulingOptions;
+  }
+
+  /**
+   * Returns the CPU platform used by this instance.
+   */
+  public String cpuPlatform() {
+    return cpuPlatform;
+  }
+
+  /**
+   * Returns a builder for the current instance.
+   */
+  public Builder toBuilder() {
+    return new BuilderImpl(this);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("id", id)
+        .add("instanceId", instanceId)
+        .add("creationTimestamp", creationTimestamp)
+        .add("description", description)
+        .add("status", status)
+        .add("statusMessage", statusMessage)
+        .add("tags", tags)
+        .add("machineType", machineType)
+        .add("canIpForward", canIpForward)
+        .add("networkInterfaces", networkInterfaces)
+        .add("attachedDisks", attachedDisks)
+        .add("metadata", metadata)
+        .add("serviceAccounts", serviceAccounts)
+        .add("schedulingOptions", schedulingOptions)
+        .add("cpuPlatform", cpuPlatform)
+        .toString();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(id, instanceId, creationTimestamp, description, status, statusMessage, tags,
+        machineType, canIpForward, networkInterfaces, attachedDisks, metadata, serviceAccounts,
+        schedulingOptions, cpuPlatform);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof InstanceInfo
+        && Objects.equals(toPb(), ((InstanceInfo) obj).toPb());
+  }
+
+  InstanceInfo setProjectId(final String projectId) {
+    Builder builder = toBuilder();
+    builder.networkInterfaces(Lists.transform(networkInterfaces,
+        new Function<NetworkInterface, NetworkInterface>() {
+          @Override
+          public NetworkInterface apply(NetworkInterface networkInterface) {
+            return networkInterface.setProjectId(projectId);
+          }
+        }));
+    builder.attachedDisks(Lists.transform(attachedDisks,
+        new Function<AttachedDisk, AttachedDisk>() {
+          @Override
+          public AttachedDisk apply(AttachedDisk attachedDisk) {
+            return attachedDisk.setProjectId(projectId);
+          }
+        }));
+    return builder.instanceId(instanceId.setProjectId(projectId))
+        .machineType(machineType.setProjectId(projectId))
+        .build();
+  }
+
+  Instance toPb() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method toPb() has an NPath complexity of 1024](https://www.codacy.com/app/mziccard/gcloud-java/file/2457017658/issues/source?bid=3140968&fileBranchId=3250778#l1214)